README.md

```diff
@@ -181,19 +185,18 @@ jobs:
           fermyon_token: ${{ secrets.FERMYON_CLOUD_TOKEN }}
 ```
 
-
-## `fermyon/action/spin/preview`
+## Deploy preview of Spin app to Fermyon Cloud - `fermyon/action/spin/preview@v1`
 
 Build and deploy the `spin` app preview to Fermyon Cloud
 
 ### Inputs
 
-| Name          | Required | Description                                                                       | Default   |
-| ------------- | -------- | --------------------------------------------------------------------------------- | --------- |
-| fermyon_token | Required | Fermyon Cloud Personal access token for deploying the `spin` app to Fermyon Cloud | -         |
-| manifest_file | Optional | Path to `spin.toml`. Used with the `build`/`deploy` command.                      | spin.toml |
-| github_token  | Required | The `GitHub` token for updating comment on PR                                     | -         |
-| undeploy      | Optional | If true, removes the preview deployment from Fermyon Cloud                        | -         |
+| Name          | Required | Default   | Description                                                                       |
+| ------------- | -------- |--------- | --------------------------------------------------------------------------------- |
+| fermyon_token | True | -         | Fermyon Cloud Personal access token for deploying the `spin` app to Fermyon Cloud |
+| manifest_file | False | spin.toml | Path to `spin.toml`. Used with the `build`/`deploy` command.                      |
+| github_token  | True | -         | The `GitHub` token for updating comment on PR                                     |
+| undeploy      | False | -         | If true, removes the preview deployment from Fermyon Cloud                        | 
```

Comment:
This raises more questions than it answers - if I set this to true, does the action do a preview deploy then undeploy?  If so, how do I see the preview?  If not, when do I use the undeploy option?  Is this undeploying a previous preview?  What if I have deployed two previews, which one does it undeploy?  What happens if undeployment fails, does it stop the PR from closing?

Reply:
may be we just remove this flag and un-deploy the preview automatically?

Reply:
That would be amazing if it were possible, but how would it work?  I would assume you'd need _some_ event after the initial preview deployment to perform the undeployment...?

Honestly, given how widely and successfully we've been using the preview action internally, I don't feel like it needs changing, it's clearly very successful the way it is.  We just need to explain this flag a little more, so that people who don't have us setting up their actions can understand how to make it work.

Reply:
let me take a first shot at it to give context, but I look forward to hearing a better way from you.

going to add the following quote before the table. 

> For triggering Github action on PR closed, add the `closed` activity type to the keyword `types`. Refer to [GitHub documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) for more details on activity types.
> 
> Then you can add `undeploy: ${{ github.event.pull_request && github.event.action == 'closed' }}` as input to the action. This will ensure `undeploy` is set to `true` only when the Github action is triggered by a PR closed event.

Reply:
so an alternative could be to just provide the info to user on how to add `closed` event as trigger and then handle closing of the Preview automatically

Reply:
I like that, and would suggest maybe just adding a bit of context around the lifecycle that we're facilitating with the deploy-undeploy behaviour.  How about something like:

> The preview lifecycle is typically to deploy a preview when a pull request is created or updated, and to remove it when the PR is closed.  To do this, create a workflow whose trigger is _(list of actions)_, and set the preview action's `undeploy` flag to _(appropriate formula)_.  The `undeploy` flag will thus resolve to `false` (deploy a preview, or update the existing preview) on PR creation or update, and to `true` (remove the preview) on PR closure.  See below for an example.
>
> If you have multiple PRs in flight, the action 'knows' which preview is associated with which PR, and will update or remove only that preview.
>
> If you don't run the preview action with `undeploy` on the closed event, your preview will remain deployed, and will count against any Fermyon Cloud account limits.

(Obviously please correct any misconceptions I have here - as you can see I am not terribly familiar with how the action works!  And other feedback welcome too of course!)

Reply:
this is perfect. thank you Ivan.